### PR TITLE
[feature-wip](multi-catalog) add iceberg tvf to read snapshots

### PR DESCRIPTION
**Pull request by wsjz**
# Proposed changes

Support new table value function `iceberg_meta("table" = "ctl.db.tbl", "query_type" = "snapshots")`
we can use the sql `select * from iceberg_meta("table" = "ctl.db.tbl", "query_type" = "snapshots")` to get snapshots info  of a table. The other iceberg metadata will be supported later when needed.

One of the usage:

Before we use following sql to time travel:
`select * from ice_table FOR TIME AS OF "2022-10-10 11:11:11"`;
`select * from ice_table FOR VERSION AS OF "snapshot_id"`;
we can use the snapshots metadata to get the `committed time` or `snapshot_id`, 
and then, we can use it as the time or version in time travel clause

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.72 seconds
 load time: 657 seconds
 storage size: 17123576208 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230104143610_clickbench_pr_73563.html

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/IcebergTableValuedFunction.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        Map<String, String> validParams = Maps.newHashMap();
+        for (String key : params.keySet()) {
+            if (!PROPERTIES_SET.contains(key.toLowerCase())) {
+                throw new AnalysisException(key + " is invalid property");
+            }
+            // check ctl db tbl
+            validParams.put(key.toLowerCase(), params.get(key));
+        }
+        String tableName = validParams.get(TABLE);
+        String queryType = validParams.get(QUERY_TYPE);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/tablefunction/IcebergTableValuedFunction.java:67]: Need to check user's privilege to this iceberg table

File: gensrc/thrift/PlanNodes.thrift [hunk 2]
```diff
@@ -55,6 +55,7 @@ enum TPlanNodeType {
   TABLE_FUNCTION_NODE,
   DATA_GEN_SCAN_NODE,
   FILE_SCAN_NODE,
+  METADATA_SCAN_NODE,
```
Comment (morningman) [inline gensrc/thrift/PlanNodes.thrift:58]: Add to the end

File: gensrc/thrift/PlanNodes.thrift [hunk 3]
```diff
@@ -370,6 +370,21 @@ struct TDataGenScanRange {
   1: optional TTVFNumbersScanRange numbers_params
 }
 
+enum TIcebergMetadataType {
+  SNAPSHOTS = 0,
+}
+
+struct TIcebergMetadataParams {
+  1: optional TIcebergMetadataType metadata_type
```
Comment (wsjz) [inline gensrc/thrift/PlanNodes.thrift:378]: move to scan node

File: be/src/vec/exec/scan/vmeta_scanner.cpp [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    auto column_size = _tuple_desc->slots().size();
+    std::vector<MutableColumnPtr> columns(column_size);
+    bool mem_reuse = block->mem_reuse();
+    do {
+        RETURN_IF_CANCELLED(state);
+
+        columns.resize(column_size);
+        for (auto i = 0; i < column_size; i++) {
+            if (mem_reuse) {
+                columns[i] = std::move(*block->get_by_position(i).column).mutate();
```
[the remark this answers is not in the document]
Comment (wsjz) [reply be/src/vec/exec/scan/vmeta_scanner.cpp:78]: keep compatibility

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**